lib/winapi/winnt.h

```diff
@@ -14,4 +14,22 @@ typedef LPCWSTR LPCTSTR;
 typedef LPCSTR LPCTSTR;
 #endif
 
+static LONG64 InterlockedExchange64(LONG64 volatile *Target, LONG64 Value)
+{
+    return __atomic_exchange_n(Target, Value, __ATOMIC_ACQ_REL);
+}
```

Comment:
Clang already has these: https://github.com/llvm-mirror/clang/blob/master/include/clang/Basic/Builtins.def#L817

So it seems that this merely needs a function or macro to be created with the right signature.

Reply:
Even then, it doesn't provide 64 bit intrinsics on x86. So this is still a lot of work. It also needs to have `_MSC_VER` defined as ugly hack to deal with clang `<intrin.h>` header oddity (it attempts to include another `intrin.h` with `#include_next`).

OpenAL-soft 1.0 does not need it (no such code). OpenAL-1.19 does need it (custom hackery using this). OpenAL-1.20 does not need it (C++ atomics).